src/directives/mmsSearch.js

```diff
@@ -102,36 +145,77 @@ function mmsSearch($window, CacheService, ElementService, ProjectService, UtilsS
 
         // event handler for multiselect metatype dropdown
         scope.multiselectEvent = {
-            onItemSelect: function(ob) {
-                $timeout( function(){
+            onItemSelect: function (ob) {
+                $timeout(function () {
                     scope.stringQueryUpdate();
-                }, 500 );
+                }, 500);
             },
-            onItemDeselect: function(ob) {
-                $timeout( function(){
+            onItemDeselect: function (ob) {
+                $timeout(function () {
                     scope.stringQueryUpdate();
-                }, 500 );
+                }, 500);
             },
-            onDeselectAll: function(ob) {
-                $timeout( function(){
+            onDeselectAll: function (ob) {
+                $timeout(function () {
                     scope.stringQueryUpdate();
-                }, 500 );
+                }, 500);
             }
         };
 
+        // Filter options
+        scope.getActiveFilterClass = function (item) {
+            if (!scope.activeFilter.length) {
+                return '';
+            }
+            return _.includes(scope.activeFilter, item);
+        };
 
-        // Watch function - TODO might not need first one
-        scope.$watchGroup(['filterQuery.query', 'facet'], function(newVal, oldVal) {
-            scope.resultFilter = {};
-            scope.resultFilter[scope.facet] = scope.filterQuery.query;
-        });
-        scope.$watch('searchResults', function(newVal) {
+        scope.filterSearchResults = function (type) {
+            var tempArr = _.clone(scope.activeFilter);
+            if (_.includes(scope.activeFilter, type) ) {
+                _.pull(scope.activeFilter, type);
+            } else {
+                scope.activeFilter.push(type);
+            }
+            _applyFilters();
+        };
+
+        var _applyFilters = function () {
+            if (!scope.activeFilter.length) {
+                scope.searchResults = baseSearchResults;
+            } else {
+                scope.searchResults = _.filter(baseSearchResults, function (item) {
+                    return _.includes(scope.activeFilter, ViewService.getElementType(item));
+                });
+            }
+        };
+
+        scope.filterOptions = [
+            { display: "Documents", icon: null, type: "Documents" },
```

Comment:
the type should match what's given back by getElementType? should be Document (singular)